Package compile error

`trait not implemented` error.  `axum-test` commit: https://github.com/JosephLenton/axum-test/commit/c2c66815a03e884c8ae820cb599ddc9c94af0be7

# repro
1. clone repo
2. `cargo test --example=example-shuttle`

## expect
Sample runs

## observe
```
$ cargo test --example=example-shuttle
   Compiling cfg-if v1.0.0
   Compiling itoa v1.0.11
   ... clip ...
   Compiling axum-extra v0.9.4
   Compiling axum-test v16.2.0 (/home/bobhy/src/rustweb/axum-test)
   Compiling axum-msgpack v0.4.0
   Compiling hyper-rustls v0.24.2
   Compiling axum v0.6.20
   Compiling hyper-timeout v0.4.1
   Compiling reqwest v0.11.27
   Compiling shuttle-common v0.48.2
   Compiling reqwest-middleware v0.2.5
   Compiling tonic v0.10.2
   Compiling shuttle-api-client v0.48.0
   Compiling shuttle-service v0.48.0
   Compiling shuttle-proto v0.48.0
   Compiling shuttle-runtime v0.48.0
   Compiling shuttle-axum v0.48.0
error[E0277]: the trait bound `Result<AxumService, shuttle_runtime::Error>: IntoTransportLayer` is not satisfied
   --> examples/example-shuttle/main.rs:72:16
    |
72  |         .build(new_app()) // <- here the application is passed in
    |          ----- ^^^^^^^^^ the trait `IntoTransportLayer` is not implemented for `Result<AxumService, shuttle_runtime::Error>`
    |          |
    |          required by a bound introduced by this call
    |
    = help: the following other types implement trait `IntoTransportLayer`:
              IntoMakeService<S>
              IntoMakeServiceWithConnectInfo<S, C>
              Router
              Serve<M, S>
              WithGracefulShutdown<M, S, F>
note: required by a bound in `TestServerBuilder::build`
   --> /home/bobhy/src/rustweb/axum-test/src/test_server_builder.rs:161:12
    |
159 |     pub fn build<A>(self, app: A) -> Result<TestServer>
    |            ----- required by a bound in this associated function
160 |     where
161 |         A: IntoTransportLayer,
    |            ^^^^^^^^^^^^^^^^^^ required by this bound in `TestServerBuilder::build`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `axum-test` (example "example-shuttle" test) due to 1 previous error
```

## second repro
I got different compile errors, also in `axum_test` in my own app.  I can't tell whether it's a similar problem.  

```
$ cargo test
   Compiling axum-test v16.2.0 (/home/bobhy/src/rustweb/axum-test)
error[E0277]: `<_ as Service<IncomingStream<'a>>>::Future` cannot be sent between threads safely
  --> /home/bobhy/src/rustweb/axum-test/src/transport_layer/into_transport_layer/into_make_service.rs:32:54
   |
32 |         let serve_handle = spawn_serve(tcp_listener, self);
   |                            -----------               ^^^^ `<_ as Service<IncomingStream<'a>>>::Future` cannot be sent between threads safely
   |                            |
   |                            required by a bound introduced by this call
   |
   = help: the trait `for<'a> Send` is not implemented for `<_ as Service<IncomingStream<'a>>>::Future`
note: this is a known limitation of the trait solver that will be lifted in the future
  --> /home/bobhy/src/rustweb/axum-test/src/transport_layer/into_transport_layer/into_make_service.rs:32:54
   |
32 |         let serve_handle = spawn_serve(tcp_listener, self);
   |                            --------------------------^^^^-
   |                            |                         |
   |                            |                         the trait solver is unable to infer the generic types that should be inferred from this argument
   |                            add turbofish arguments to this call to specify the types manually, even if it's redundant
note: required by a bound in `util::spawn_serve::spawn_serve`
  --> /home/bobhy/src/rustweb/axum-test/src/util/spawn_serve.rs:20:57
   |
17 | pub fn spawn_serve<M, S>(tcp_listener: TcpListener, make_service: M) -> ServeHandle
   |        ----------- required by a bound in this function
...
20 |     for<'a> <M as Service<IncomingStream<'a>>>::Future: Send,
   |                                                         ^^^^ required by this bound in `spawn_serve`

error[E0277]: `<_ as Service<IncomingStream<'a>>>::Future` cannot be sent between threads safely
  --> /home/bobhy/src/rustweb/axum-test/src/transport_layer/into_transport_layer/into_make_service_with_connect_info.rs:30:54
   |
30 |         let serve_handle = spawn_serve(tcp_listener, self);
   |                            -----------               ^^^^ `<_ as Service<IncomingStream<'a>>>::Future` cannot be sent between threads safely
   |                            |
   |                            required by a bound introduced by this call
   |
   = help: the trait `for<'a> Send` is not implemented for `<_ as Service<IncomingStream<'a>>>::Future`
note: this is a known limitation of the trait solver that will be lifted in the future
  --> /home/bobhy/src/rustweb/axum-test/src/transport_layer/into_transport_layer/into_make_service_with_connect_info.rs:30:54
   |
30 |         let serve_handle = spawn_serve(tcp_listener, self);
   |                            --------------------------^^^^-
   |                            |                         |
   |                            |                         the trait solver is unable to infer the generic types that should be inferred from this argument
   |                            add turbofish arguments to this call to specify the types manually, even if it's redundant
note: required by a bound in `util::spawn_serve::spawn_serve`
  --> /home/bobhy/src/rustweb/axum-test/src/util/spawn_serve.rs:20:57
   |
17 | pub fn spawn_serve<M, S>(tcp_listener: TcpListener, make_service: M) -> ServeHandle
   |        ----------- required by a bound in this function
...
20 |     for<'a> <M as Service<IncomingStream<'a>>>::Future: Send,
   |                                                         ^^^^ required by this bound in `spawn_serve`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `axum-test` (lib) due to 2 previous errors

```
## additional info

Rust version:
```
$ rustup -V
rustup 1.27.1 (54dd3d00f 2024-04-24)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.81.0 (eeb90cda1 2024-09-04)`
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,6 +16,21 @@ readme = "README.md"
 all-features = true
 rustdoc-args = ["--cfg", "docsrs"]
 
+[[example]]
+name = "example-shuttle"
+path = "examples/example-shuttle/main.rs"
+required-features = ["shuttle"]
+
+[[example]]
+name = "example-websocket-chat"
+path = "examples/example-websocket-chat/main.rs"
+required-features = ["ws"]
+
+[[example]]
+name = "example-websocket-ping-pong"
+path = "examples/example-websocket-ping-pong/main.rs"
+required-features = ["ws"]
+
 [features]
 default = ["pretty-assertions"]
 

